upgrade/util.go

```diff
@@ -38,6 +38,8 @@ type Context struct {
 	UpgradeProviderVersion bool
 	MajorVersionBump       bool
 
+	UpgradeJavaVersion bool
```

Comment:
Why do we have to opt into it? Why not upgrade it every time we update codegen?

Reply:
It currently defaults to true so in effect it does upgrade every time unless explicitly set to false. I also imagined it would give us a more flexible safeguard in case we ever have a bad pulumi-java version and want to halt upgrades.

But you're right that we can likely do without.

Reply:
Ahh that's nice. That's great then.